### PR TITLE
modify role to clusterrole

### PR DESCRIPTION
**Pull request by innerpeacez**

### CODE REVIEW

**Comment by innerpeacez** [comment]
@wu-sheng @hanahmily  help review

**Comment by hanahmily** [comment]
Most of it looks great. Thanks @innerpeacez. We leverage oap permission here is because envoy recevier intends to access API server get information across the entire cluster. So a more fine grain method is to identify whether user opens this receiver then grant oap a appropriate permission. Well, this way is much simpler for our user. Can you document it in README. md

**Comment by wu-sheng** [comment]
> Most of it looks great. Thanks @innerpeacez. We leverage oap permission here is because envoy recevier intends to access API server get information across the entire cluster. So a more fine grain method is to identify whether user opens this receiver then grant oap a appropriate permission. Well, this way is much simpler for our user. Can you document it in README. md

@innerpeacez I think env varible description, helm guide and API server grant, should be added in this doc, https://github.com/apache/skywalking/blob/master/docs/en/setup/envoy/als_setting.md#observe-service-mesh-through-als

**Comment by wu-sheng** [comment]
Change the document step 2-3 and guide the user to use the helm, please

**Comment by innerpeacez** [comment]
I will add a `oap.envoy.sla.enabled` key to the values.yaml file to control whether envoy sla is turned on. And when turned on, clusterrole will be used。 @hanahmily @wu-sheng

**Comment by innerpeacez** [comment]
@hanahmily @wu-sheng Help review again,thanks .

**Comment by innerpeacez** [comment]
If there are no problems, I will merge this PR

File: helm-chart/helm3/6.4.0/skywalking/templates/oap-deployment.yaml [hunk 1]
```diff
@@ -91,6 +91,10 @@ spec:
               fieldPath: metadata.uid
         - name: SW_STORAGE
           value: elasticsearch
+        {{- if .Values.oap.envoy.als.enabled }}
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/templates/oap-deployment.yaml]: why do you add a parameter to control an env variable? all items listed in `.Values.oap.env` will be added here. Take a look at line 100.

File: helm-chart/helm3/6.4.0/skywalking/README.md [hunk 2]
```diff
@@ -111,7 +112,7 @@ The following table lists the configurable parameters of the Skywalking chart an
 | `elasticsearch.master.tolerations` | `[]` | Master tolerations |
 | `elasticsearch.master.heapSize` | `512m` | Master node heap size |
 | `elasticsearch.master.name` | `master` | Master component name |
-| `elasticsearch.master.persistence.enabled` | `true` | Master persistent enabled/disabled |
+| `elasticsearch.master.persistence.enabled` | `false` | Master persistent enabled/disabled |
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/README.md]: Why disable all of the persistence?

Comment (innerpeacez) [reply helm-chart/helm3/6.4.0/skywalking/README.md]: If the storageClass is not set, an error will be returned.

File: helm-chart/helm3/6.4.0/skywalking/templates/oap-deployment.yaml (hunk 1 above)
Comment (innerpeacez) [reply helm-chart/helm3/6.4.0/skywalking/templates/oap-deployment.yaml]: In order to enable envoy als, there is a clusterrole privilege, and when it is not enabled, only role privilege

File: helm-chart/helm3/6.4.0/skywalking/README.md (hunk 2 above)
Comment (hanahmily) [reply helm-chart/helm3/6.4.0/skywalking/README.md]: 👍

File: helm-chart/helm3/6.4.0/skywalking/README.md [hunk 3]
```diff
@@ -196,3 +197,18 @@ ui:
...
+### Envoy ALS
+
+Envoy ALS(access log service) provides fully logs about RPC routed, including HTTP and TCP.
+
+If you want to open envoy sla, you can do this by modifying values.yaml. 
+
+```yaml
+oap:
+  envoy:
+    sla:
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/README.md]: typo. should be `als`

File: helm-chart/helm3/6.4.0/skywalking/README.md [hunk 4]
```diff
@@ -196,3 +197,18 @@ ui:
...
+      enabled: true
+```
+
+When envoy sla , will give ServerAccount clusterrole permission.
+More envoy sla ,please refer to https://github.com/apache/skywalking/blob/master/docs/en/setup/envoy/als_setting.md#observe-service-mesh-through-als
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/README.md]: from `sla` to `als`

File: helm-chart/helm3/6.4.0/skywalking/README.md [hunk 5]
```diff
@@ -196,3 +197,18 @@ ui:
         hosts:
           - skywalking.domain.com
 ```
+### Envoy ALS
+
+Envoy ALS(access log service) provides fully logs about RPC routed, including HTTP and TCP.
+
+If you want to open envoy sla, you can do this by modifying values.yaml.
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/README.md]: should be `als`

Comment (hanahmily) [reply helm-chart/helm3/6.4.0/skywalking/README.md]: and should be upper-case

File: helm-chart/helm3/6.4.0/skywalking/README.md [hunk 6]
```diff
@@ -56,6 +56,7 @@ The following table lists the configurable parameters of the Skywalking chart an
 | `oap.nodeSelector`                    | OAP labels for master pod assignment                               | `{}`                                |
 | `oap.tolerations`                     | OAP tolerations                                                    | `[]`                                |
 | `oap.resources`                       | OAP node resources requests & limits                               | `{} - cpu limit must be an integer` |
+| `oap.envoy.sla.enabled`               | Open envoy sla                                                     | `false`                             |
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/README.md]: `als`

File: helm-chart/helm3/6.4.0/skywalking/values.yaml [hunk 7]
```diff
@@ -31,9 +31,13 @@ oap:
     #   memory: 4Gi
   # podAnnotations:
   #   example: oap-foo
+  envoy:
+    als:
+      enabled: false
+      # more envoy sla ,please refer to https://github.com/apache/skywalking/blob/master/docs/en/setup/envoy/als_setting.md#observe-service-mesh-through-als
```
Comment (hanahmily) [inline helm-chart/helm3/6.4.0/skywalking/values.yaml]: should be `ALS`

**Comment by hanahmily** [comment]
@innerpeacez after some typo is fixed I think it's fine to merge it.

**Pull request merged**